barretenberg/cpp/src/barretenberg/vm/tests/AvmMini_bitwise.test.cpp

```diff
@@ -0,0 +1,244 @@
+
+    EXPECT_EQ(alu_row.aluChip_alu_u8_tag, FF(1));
+    validate_trace_proof(std::move(trace));
+}
+
+TEST_F(AvmMiniBitwiseTestsU16, BitwiseNot)
+{
+    // trace_builder
+    uint16_t input = 1;
+    trace_builder.set(input, 0, AvmMemoryTag::U16);     // Memory Layout: [1,0,0,...]
+    trace_builder.bitwise_not(0, 1, AvmMemoryTag::U16); // [1,0,255,0,0,....]
+    trace_builder.return_op(1, 1);
+    auto trace = trace_builder.finalize();
+
+    uint16_t res = ~input;
+    auto alu_row = common_validate_bitwise_not(trace, FF(1), FF(res), FF(0), FF(1), AvmMemoryTag::U16);
+
+    EXPECT_EQ(alu_row.aluChip_alu_u16_tag, FF(1));
+    validate_trace_proof(std::move(trace));
+}
+
+TEST_F(AvmMiniBitwiseTestsU32, BitwiseNot)
+{
+    // trace_builder
+    uint32_t input = 1;
+    trace_builder.set(input, 0, AvmMemoryTag::U32);     // Memory Layout: [1,0,0,...]
+    trace_builder.bitwise_not(0, 1, AvmMemoryTag::U32); // [1,0,255,0,0,....]
+    trace_builder.return_op(1, 1);
+    auto trace = trace_builder.finalize();
+
+    uint32_t res = ~input;
+    auto alu_row = common_validate_bitwise_not(trace, FF(1), FF(res), FF(0), FF(1), AvmMemoryTag::U32);
+
+    EXPECT_EQ(alu_row.aluChip_alu_u32_tag, FF(1));
+    validate_trace_proof(std::move(trace));
+}
+
+TEST_F(AvmMiniBitwiseTestsU64, BitwiseNot)
+{
+    // trace_builder
+    uint64_t input = 1;
+    trace_builder.set(input, 0, AvmMemoryTag::U64);     // Memory Layout: [1,0,0,...]
+    trace_builder.bitwise_not(0, 1, AvmMemoryTag::U64); // [1,0,255,0,0,....]
+    trace_builder.return_op(1, 1);
+    auto trace = trace_builder.finalize();
+
+    uint64_t res = ~input;
+    auto alu_row = common_validate_bitwise_not(trace, FF(1), FF(res), FF(0), FF(1), AvmMemoryTag::U64);
+
+    EXPECT_EQ(alu_row.aluChip_alu_u64_tag, FF(1));
+    validate_trace_proof(std::move(trace));
+}
+
+TEST_F(AvmMiniBitwiseTestsU128, BitwiseNot)
+{
+    // trace_builder
+    uint128_t input = 1;
+    trace_builder.set(input, 0, AvmMemoryTag::U128);     // Memory Layout: [1,0,0,...]
+    trace_builder.bitwise_not(0, 1, AvmMemoryTag::U128); // [1,0,255,0,0,....]
+    trace_builder.return_op(1, 1);
+    auto trace = trace_builder.finalize();
+
+    uint128_t res = ~input;
+    auto alu_row =
+        common_validate_bitwise_not(trace, FF(1), FF(uint256_t::from_uint128(res)), FF(0), FF(1), AvmMemoryTag::U128);
+
+    EXPECT_EQ(alu_row.aluChip_alu_u128_tag, FF(1));
+    validate_trace_proof(std::move(trace));
+}
+
+/******************************************************************************
+ *
+ * NEGATIVE TESTS - Finite Field Type
+ *
+ ******************************************************************************
+ * See AvmMini_arithmetic.cpp for explanation of negative tests
+ ******************************************************************************/
+
+std::vector<Row> gen_mutated_trace_not(FF const& a, FF const& c_mutated, avm_trace::AvmMemoryTag tag)
```

Comment:
Please move this helper routine to the anonymous namespace at the beginning of the file.